app/models/rubric_criterion.rb

```diff
@@ -1,54 +1,49 @@
 class RubricCriterion < Criterion
-  before_validation :scale_marks_if_max_mark_changed
   before_save :round_max_mark
 
   validates :levels, presence: true
 
   DEFAULT_MAX_MARK = 4
   # Checks whether the passed in param's level_attributes have unique name and marks.
   # Skips the uniqueness validation if true.
-  def update_levels(levels_attributes)
+  def update_levels(levels_attributes, scale = 1.0)
     s1, s2 = Set[], Set[]
     should_skip = true
 
-    # Check if there's a dup in marks or names
     levels_attributes.each_value do |val|
-      if s1.include?(val[:mark]) || s2.include?(val[:name])
-        should_skip = false
-        break
+      # Check if there's a dup in marks or names
+      unless should_skip
+        if s1.include?(val[:mark]) || s2.include?(val[:name])
+          should_skip = false
+        end
+        s1.add(val[:mark])
+        s2.add(val[:name])
+      end
+
+      # if we're updating an existing mark
+      if val.key?(:id)
+        existing_level = self.levels.find(val[:id])
+
+        # only scale if the existing mark is the same as the incoming mark
+        if val[:mark].to_f.round(2) == existing_level.mark.round(2)
+          val[:mark] = (val[:mark].to_f * scale).round(2).to_s
+        end
       end
-      s1.add(val[:mark])
-      s2.add(val[:name])
     end
 
     if should_skip
       levels_attributes.each_value do |val|
         val[:skip_uniqueness_validation] = true
       end
     end
+
     self.update(levels_attributes: levels_attributes)
   end
 
   def level_with_mark_closest_to(mark)
     self.levels.min_by { |m| (m.mark - mark).abs }
   end
 
-  def scale_marks_if_max_mark_changed
```

Comment:
Decided to do this manually, from my obeservations this wasn't getting triggered except on update anyways.

Reply:
You shouldn't delete this; there may be other ways of updating `RubricCriterion` other than the controller method you're working on.

Instead, you can *skip* this validation in the controller method when calling `update` on the criterion.